src/Sylius/Component/Promotion/Model/Promotion.php

```diff
@@ -416,8 +357,6 @@ public function removeRule(RuleInterface $rule)
     {
         $rule->setPromotion(null);
```

Comment:
These changes are not related to this PR and that change should be discussed separately, same as all below.